4_evaluation/src/evaluate_models.R

```diff
@@ -0,0 +1,79 @@
+## Creating function to use the metrics function for future input into the Delaware project pipeline:
+
+# Calls the four metrics functions (mae, rmse, mare, and nash). This function accepts 2 arguments data-in and grouping, the inputted data will be grouped by model to suit the data we're working with. The grouping arguments accepts 'NA' for no grouping, or any other grouping such as by segment, month, year, etc. We call the four metrics function to produce a dataframe with metrics of interest.
+calc_all_metric <- function(file_path, dat_in, grouping) {
+  dat_in <- dat_in %>%
+    mutate(year = lubridate::year(date),
+           month = lubridate::month(date)) %>%
+    group_by(model)
+
+  group_test <- deparse(substitute(grouping))
+  if (group_test == 'NA') {
+    dat_mod <- dat_in
+  } else {
+    dat_mod <- dat_in %>%
+      group_by(model, .data[[grouping]])
+  }
+
+  metrics <- dat_mod %>%
+    dplyr::summarize(n = n(),
+              mae = calc_mae(observe_data = temp_c,
+                             predict_data = predicted),
+              rmse =  calc_rmse(observe_data = temp_c,
+                                predict_data = predicted),
+              mare = calc_mare(observe_data = temp_c,
+                               predict_data = predicted),
+              nse = calc_nash(observe_col = temp_c,
+                              predict_col =  predicted))
+  write_csv(metrics, path = file_path)
+  return(file_path)
```

Comment:
You don't need returns when you are writing a file. You can delete these for all of these functions that are writing csvs.